packages/lib/src/components/internal/QRLoader/QRLoader.tsx

```diff
@@ -19,251 +18,238 @@ import { AnalyticsInfoEvent } from '../../../core/Analytics/AnalyticsInfoEvent';
-            });
+            if (actualTimePassed >= props.throttleTime && currentDelay !== props.throttledInterval) {
+                setDelay(props.throttledInterval);
+                currentDelay = props.throttledInterval;
+            }
         };
 
-        const finalState = (image: string, message: string) => {
-            const status = i18n.get(message);
-            useA11yReporter(status);
-            return (
-                <div className="adyen-checkout__qr-loader adyen-checkout__qr-loader--result">
-                    <img
-                        className="adyen-checkout__qr-loader__icon adyen-checkout__qr-loader__icon--result"
-                        src={getImage({ imageFolder: 'components/' })(image)}
-                        alt={status}
-                    />
-                    <div className="adyen-checkout__qr-loader__subtitle">{status}</div>
-                </div>
-            );
+        timeoutRef.current = setTimeout(() => {
+            void statusInterval();
+        }, currentDelay);
+
+        return () => {
+            clearTimeout(timeoutRef.current);
         };
+    }, [expired, completed, loading, delay, props.throttleTime, timePassed]);
 
-        if (expired) {
-            return finalState('error', 'error.subtitle.payment');
-        }
+    const { amount, showAmount, url, brandLogo, brandName, countdownTime, type, onActionHandled } = props;
 
-        if (completed) {
-            return finalState('success', 'creditCard.success');
-        }
+    const qrCodeImage = props.qrCodeData ? `${loadingContext}${QRCODE_URL}${props.qrCodeData}&clientKey=${props.clientKey}` : props.qrCodeImage;
 
-        if (loading) {
-            return (
-                <div className="adyen-checkout__qr-loader">
-                    {brandLogo && (
-                        <div className="adyen-checkout__qr-loader__brand-logo-wrapper">
-                            <img alt={brandName} src={brandLogo} className="adyen-checkout__qr-loader__brand-logo" />
-                        </div>
-                    )}
-                    <Spinner />
-                </div>
-            );
-        }
+    const handleCopy = (complete: () => void) => {
+        void copyToClipboard(props.qrCodeData);
+
+        const event = new AnalyticsInfoEvent({
+            type: ANALYTICS_DOWNLOAD_STR,
+            target: ANALYTICS_QR_CODE_DOWNLOAD
+        });
+        props.onSubmitAnalytics(event);
+        complete();
+    };
 
-        const qrSubtitleRef = useAutoFocus();
-        const classnames = this.props.classNameModifiers.map(m => `adyen-checkout__qr-loader--${m}`);
+    const onQrCodeLoad = () => {
+        onActionHandled?.({
+            componentType: props.type,
+            actionDescription: 'qr-code-loaded'
+        });
+    };
 
+    const finalState = (image: string, message: string) => {
+        const status = i18n.get(message);
+        useA11yReporter(status);
```

Comment:
React hooks shouldn't be used conditionally. I see that this is not a 'new' change to it, but this can lead to misbehavior if I am not mistaken.

Could that be moved to the top of the Component?